library/std/src/fs.rs

```diff
@@ -2803,8 +2803,8 @@ pub fn create_dir<P: AsRef<Path>>(path: P) -> io::Result<()> {
 /// Recursively create a directory and all of its parent components if they
 /// are missing.
 ///
-/// If this function returns an error, some of the parent components might have
-/// been created already.
```

Comment:
The old and the new paragraph are about completely different things. The old one is still correct, if an error is encountered it may still have created some of the intermediate directories, i.e. it's not atomic.
E.g. some intermediate path component contains an invalid character or the filesystem becomes full just in the middle of the operation or something like that.

Reply:
Ok, I thought it says that an error can be returned if some of the parent directories already exist. Perhaps it can be reworded better? For example,

> This function is not atomic. In the event of an error, the intermediate parent components that were created will persist.